src/BinDeps.jl

```diff
@@ -194,7 +194,7 @@ end
 
 type Choices <: BuildStep
     choices::Vector{Choice}
-    Choices() = new(Array(Choice,0))
+    Choices() = new(Vector{Choice}())
```

Comment:
there's some discussion that `Vector{T}()` might get deprecated - hasn't happened yet, but evidently the zero arg form isn't loved by all